### ISSUE TITLE
Improve S2198 (C#): Include boundary checks

### ISSUE THREAD

**Issue by gregory-paidis-sonarsource** [SonarSource/sonar-dotnet#6735]
Right now, S2198 implements **only** out-of-range checks for C#, but for other languages it also includes checks at the exact boundary.

Example:

```csharp
const int bigBoundary = int.MaxValue;
const int smallBoundary = int.MinValue;

var x = 42;

var alwaysSmallerOrEqual = x <= bigBoundary; // Always true, should raise
var alwaysBiggerOrEqual = x >= smallBoundary; // Always true, should raise

```

We should also implement the boundary check in C# for all the integral types (plus `char`), to be consistent with the other languages.

**Comment by pavel-mikula-sonarsource** [comment]
Is it a SE rule on other languages?

**Comment by pavel-mikula-sonarsource** [comment]
To see if we need to turn it into a SE rule or not. 
If it's not SE for Java, our `FindConstantValue` method should probably be good enough to determine the value we compare against.

**Comment by pavel-mikula-sonarsource** [comment]
Moved to [NET-1782](https://sonarsource.atlassian.net/browse/NET-1782)

[NET-1782]: https://sonarsource.atlassian.net/browse/NET-1782?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
